pinot-core/src/main/java/org/apache/pinot/core/common/ColumnValueReader.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.common;
+
+import org.apache.pinot.core.io.reader.ForwardIndexReader;
+import org.apache.pinot.core.io.reader.ReaderContext;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+
+
+/**
+ * ColumnValueReader is a wrapper class on top of the forward index reader to read values from it.
+ * <p>Batch value read APIs support the following data type conversions for numeric types:
+ * <ul>
+ *   <li>INT -> LONG</li>
+ *   <li>INT, LONG -> FLOAT</li>
+ *   <li>INT, LONG, FLOAT -> DOUBLE</li>
+ * </ul>
+ */
+@SuppressWarnings({"rawtypes", "unchecked"})
+public final class ColumnValueReader {
+  private final ForwardIndexReader _forwardIndexReader;
```

Comment:
Should this extend forwardIndexReader, as it seems to be pass-through for several api's, and providing additional functionality?

Reply:
This is a wrapper class on top of the forward index to help manage the reader context. It replaces the current SingleValueReader and MultiValueReader. It should not implement ForwardIndexReader but access forward index.